### ISSUE TITLE
Bad italics with underscores within word

### ISSUE THREAD

**Issue by eerohele** [yogthos/markdown-clj#180]
Repro:

```clojure
(markdown.core/md-to-html-string "foo_bar_baz")
;;=> "<p>foo<i>bar</i>baz</p>"
```

Expected:

```clojure
(markdown.core/md-to-html-string "foo_bar_baz")
;;=> "<p>foo_bar_baz</p>"
```

For reference, see [this commonmark.js dingus](https://spec.commonmark.org/dingus/?text=foo_bar_baz).

**Comment by yogthos** [comment]
I'd be open to making a change to check for space before and after, as a workaround you can escape the italics with `\_` at the moment.

**Comment by michihuber** [comment]
Could this be an option to be passed to `make-separator`, to require spaces before and after? this is also problematic in the context of urls (not links) that are in the text.

**Comment by yogthos** [comment]
Yeah, passing a an option would work, it could be added in the config options when calling the parser and then `make-separator` could check for that.
